SIPS/sip-29.md

```diff
@@ -243,13 +266,138 @@ type FungibleAssetMetadata = {
   // List of asset units.
   units: FungibleAssetUnit[];
 };
+```
+### Asset Market Data
+
+```typescript
+type FungibleAssetMarketData = {
+  // Represents a fungible asset market data.
+  fungible: true;
+
+  // The market cap of the asset represented as a decimal number in a string.
+  marketCap?: string;
+
+  // The total volume of the asset represented as a decimal number in a string.
+  totalVolume?: string;
+
+  // The circulating supply of the asset represented as a decimal number in a string.
+  circulatingSupply?: string;
+
+  // The all time high of the asset represented as a decimal number in a string.
+  allTimeHigh?: string;
+
+  // The all time low of the asset represented as a decimal number in a string.
+  allTimeLow?: string;
+
+  pricePercentChange?: {
+    // The `all` value is a special interval that represents all available data.  
+    all?: number;  
+
+    // The interval key MUST follow the ISO 8601 duration format.  
+    [interval: string]: number;
+  };
+};
+```
+
+## Appendix II: Non-fungible Asset Metadata & Market Data
+
+The following asset metadata fields for a non-fungible asset are defined.
+
+### Asset Metadata
+
+`Caip10Address` - A string that represents an address using the [CAIP-10][caip-10] standard.
+
+```typescript
+type NonFungibleAssetCollection = {
+  // Human-friendly name of the asset collection.
+  name: string;
+
+  // Ticker symbol of the asset collection.
+  symbol: string;
+
+  // The number of tokens in the collection.
+  tokenCount: string;
+
+  // The creator address of the asset.
+  creator?: Caip10Address;
+
+  // Base64 data URI or URL representation of the asset icon.
+  imageUrl?: string;
+};
+
+type NonFungibleAssetMetadata = {
```

Comment:
This type should probably have a `fungible: false` flag too.

Reply:
https://github.com/MetaMask/SIPs/blob/a30f9a0234dd67628a39fe997260035efd440984/SIPS/sip-29.md?plain=1#L345

Reply:
We have both :thinking: